### PR TITLE
ROCANA-6477 Add Process IO metrics and unit tests for Linux

### PR DESCRIPTION
**Pull request by actgardner**
Add unit tests for all the process metrics, and IO metrics for each process

### CODE REVIEW

**Comment by adamlamar** [comment]
The tests fail for me:

```
sigarLinux
/home/vagrant/scaling-data/go/src/github.com/scalingdata/gosigar/sigar_linux_test.go:607
  Process
  /home/vagrant/scaling-data/go/src/github.com/scalingdata/gosigar/sigar_linux_test.go:606
    GetsProcessTime [It]
    /home/vagrant/scaling-data/go/src/github.com/scalingdata/gosigar/sigar_linux_test.go:558

    Expected
        <uint64>: 1466308497000
    to equal
        <uint64>: 1465314458000

    /home/vagrant/scaling-data/go/src/github.com/scalingdata/gosigar/sigar_linux_test.go:557
```

File: sigar_linux_test.go [hunk 1]
```diff
@@ -490,4 +490,118 @@ Inter-|   Receive                                                |  Transmit
...
+			Expect(len(procList.List)).To(Equal(1))
+			Expect(procList.List[0]).To(Equal(10))
+		})
+
+		It("GetsArgs", func() {
+			cmdLineFile := procd + "/10/cmdline"
+			cmdLine := "/sbin/min\x00getty\x00/dev/tty3\x00"
+			err := os.MkdirAll(procd+"/10/", 0777)
+			Expect(err).ToNot(HaveOccurred())
+			err = ioutil.WriteFile(cmdLineFile, []byte(cmdLine), 0444)
```
Comment (adamlamar) [inline sigar_linux_test.go:514]: Missing a `Expect(err).ToNot(HaveOccurred())`?

Comment (adamlamar) [inline sigar_linux_test.go:514]: It looks like all the usages of `ioutil.WriteFile` are missing an assertion. Was leaving it off unintentional?

Comment (actgardner) [inline sigar_linux_test.go:514]: Nope, looks like I missed that assertion. Good catch!

**Comment by actgardner** [comment]
Time zones!

**Comment by actgardner** [comment]
@adamlamar Can you give this a second pass? The issue actually wasn't timezones, we weren't stubbing out the start time of the machine.

**Comment by adamlamar** [comment]
Sorry @alanctgardner, still get a (different) test failure:

```
• Failure [0.023 seconds]
sigarLinux
/home/vagrant/scaling-data/go/src/github.com/scalingdata/gosigar/sigar_linux_test.go:619
  Process
  /home/vagrant/scaling-data/go/src/github.com/scalingdata/gosigar/sigar_linux_test.go:618
    GetsProcessTime [It]
    /home/vagrant/scaling-data/go/src/github.com/scalingdata/gosigar/sigar_linux_test.go:567

    Expected
        <uint64>: 0
    to equal
        <uint64>: 2000

    /home/vagrant/scaling-data/go/src/github.com/scalingdata/gosigar/sigar_linux_test.go:564
------------------------------
```

**Comment by actgardner** [comment]
@adamlamar Third time's a charm?

File: sigar_linux_test.go [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package sigar_test
 
 import (
+	"fmt"
```
Comment (adamlamar) [inline sigar_linux_test.go]: unused import

**Comment by adamlamar** [comment]
I get the unused import for `fmt`, but after removing it, the tests pass! +1

**Comment by actgardner** [comment]
Usually I'm better at writing code than this, I swear ;)

**Pull request merged**